### PR TITLE
Use the fully qualified domain name for getent calls

### PR DESCRIPTION
**Pull request by frankgh**

### CODE REVIEW

File: concourse/scripts/install_pxf.bash [hunk 1]
```diff
@@ -11,10 +11,10 @@ REALM_2=${REALM_2:-}
 KERBEROS=${KERBEROS:-false}
 if [[ -f terraform_dataproc/name ]]; then
 	HADOOP_HOSTNAME=ccp-$(< terraform_dataproc/name)-m
-	HADOOP_IP=$(getent hosts "$HADOOP_HOSTNAME" | awk '{ print $1 }')
+	HADOOP_IP=$(getent hosts "${HADOOP_HOSTNAME}.c.data-gpdb-ud.internal" | awk '{ print $1 }')
 elif [[ -f dataproc_env_files/name ]]; then
 	HADOOP_HOSTNAME=$(< dataproc_env_files/name)
-	HADOOP_IP=$(getent hosts "$HADOOP_HOSTNAME" | awk '{ print $1 }')
+	HADOOP_IP=$(getent hosts "${HADOOP_HOSTNAME}.c.data-gpdb-ud.internal" | awk '{ print $1 }')
```
Comment (oliverralbertini) [inline concourse/scripts/install_pxf.bash]: Is the value `.c.data-gpdb-ud.internal` hard-coded in other places, or do we parametrize it?

File: concourse/scripts/load_data.bash [hunk 2]
```diff
@@ -15,10 +15,10 @@ set -x
...
 HADOOP_HOSTNAME="ccp-$(cat terraform_dataproc/name)-m"
 
-gcloud compute instances add-metadata ${HADOOP_HOSTNAME} \
+gcloud compute instances add-metadata "${HADOOP_HOSTNAME}" \
   --metadata ssh-keys="ccp-ci-service:$(cat ~/.ssh/google_compute_engine.pub)" \
   --zone "us-central1-a"
 
 ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no \
-  -t -i ~/.ssh/google_compute_engine ccp-ci-service@${HADOOP_HOSTNAME} \
+  -t -i ~/.ssh/google_compute_engine ccp-ci-service@"${HADOOP_HOSTNAME}.c.data-gpdb-ud.internal" \
```
Comment (oliverralbertini) [inline concourse/scripts/load_data.bash]: bash style nit pick: start the quote at the beginning of the word, unless it contains `~` or other shell characters that change meaning inside quotes:

`"ccp-ci-service@${HADOOP_HOSTNAME}.c.data-gpdb-ud.internal"`

File: server/pxf-api/src/main/java/org/greenplum/pxf/api/model/BaseConfigurationFactory.java [hunk 3]
```diff
@@ -70,7 +70,7 @@ public Configuration initConfiguration(String configDirectory, String serverName
         }
 
         if (ArrayUtils.isEmpty(serverDirectories)) {
-            LOG.warn("Directory {}{}{} does not exist or cannot be read by PXF, no configuration resources are added for server {}",
+            LOG.debug("Directory {}{}{} does not exist or cannot be read by PXF, no configuration resources are added for server {}",
```
Comment (oliverralbertini) [inline server/pxf-api/src/main/java/org/greenplum/pxf/api/model/BaseConfigurationFactory.java]: Is this related to the above changes?

File: concourse/scripts/test_pxf_multinode.bash [hunk 4]
```diff
@@ -150,6 +150,9 @@ function setup_pxf_kerberos_on_cluster() {
 	scp centos@mdw:/etc/krb5.conf /tmp/krb5.conf
 	sudo cp /tmp/krb5.conf /etc/krb5.conf
 
+	# Add foreign dataproc hostfile to /etc/hosts
+	sudo tee --append /etc/hosts < "${DATAPROC_DIR}"/etc_hostfile
```
Comment (oliverralbertini) [inline concourse/scripts/test_pxf_multinode.bash]: same comment as above:
`"${DATAPROC_DIR}/etc_hostfile"`

File: concourse/scripts/install_pxf.bash (hunk 1 above)
Comment (frankgh) [reply concourse/scripts/install_pxf.bash]: it's mostly hardcoded, we should parameterize it.

File: server/pxf-api/src/main/java/org/greenplum/pxf/api/model/BaseConfigurationFactory.java (hunk 3 above)
Comment (frankgh) [reply server/pxf-api/src/main/java/org/greenplum/pxf/api/model/BaseConfigurationFactory.java]: hmm, nope, i was seeing annoying log entries when running pxf_fdw, maybe I should remove it as it is unrelated

File: concourse/scripts/install_pxf.bash [hunk 5]
```diff
@@ -8,13 +8,14 @@ GPHOME=/usr/local/greenplum-db-devel
 MASTER_HOSTNAME=$(grep < cluster_env_files/etc_hostfile '\bmdw' | awk '{print $2}')
 REALM=${REALM:-}
 REALM_2=${REALM_2:-}
+GOOGLE_PROJECT_ID=${GOOGLE_PROJECT_ID:-data-gpdb-ud}
 KERBEROS=${KERBEROS:-false}
 if [[ -f terraform_dataproc/name ]]; then
 	HADOOP_HOSTNAME=ccp-$(< terraform_dataproc/name)-m
-	HADOOP_IP=$(getent hosts "$HADOOP_HOSTNAME" | awk '{ print $1 }')
+	HADOOP_IP=$(getent hosts "${HADOOP_HOSTNAME}.c.${GOOGLE_PROJECT_ID}.internal" | awk '{ print $1 }')
```
Comment (denalex) [inline concourse/scripts/install_pxf.bash]: what is the magic of finding the IP address here, when hosts are not in /etc/hosts ? Google'
s implementation knows their own internal DNS ? Why was it working before and changed ?

File: server/pxf-api/src/main/java/org/greenplum/pxf/api/model/BaseConfigurationFactory.java (hunk 3 above)
Comment (oliverralbertini) [reply server/pxf-api/src/main/java/org/greenplum/pxf/api/model/BaseConfigurationFactory.java]: That's OK, just wasn't sure if I was missing something.

File: concourse/scripts/install_pxf.bash (hunk 5 above)
Comment (oliverralbertini) [reply concourse/scripts/install_pxf.bash]: that's the mystery. we suspect it's related to updating Concourse.

Comment (frankgh) [reply concourse/scripts/install_pxf.bash]: it's detailed here: https://cloud.google.com/compute/docs/internal-dns

Comment (frankgh) [reply concourse/scripts/install_pxf.bash]: we don't know what changed, but by providing the full internal DNS we are able to solve the issue
